Import script: Fix bugs.

rheaparekh opened the pull request:
Fixes #8949 and #8959

timabbott commented on zerver/lib/export.py:
```diff
@@ -1461,6 +1462,8 @@ def import_uploads_local(import_dir: Path, processing_avatars: bool=False,
             if record['s3_path'].endswith('.original'):
                 user_profile = get_user_profile_by_id(record['user_profile_id'])
+                if os.path.exists(medium_file_path):
+                    os.remove(medium_file_path)
```
This doesn't work, because of how variable scoping works in Python; `medium_file_path` needs to be set in the same loop iteration where it's used.

timabbott commented:
I merged the first 2 commits; see the comment above for the third.

rheaparekh commented:
@timabbott I had corrected this, but I pushed the wrong commit here in a hurry. Thanks for the sanity check. I have updated this and I'll be careful next time.

timabbott commented:
Merged the last commit as ed7127c8b4393549c593c78abd37fa4b63d26ff1, thanks @rheaparekh!  

It's important when submitting a fix for a subtle issue like this to explain clearly the problem being fixed, not just what the solution is.  Comments and the commit history can be really valuable to future developers (including our own future selves) for understanding why the code is the way it is, but only if we actually explain subtle problems like this that motivated the changes.

rheaparekh commented:
@timabbott Thankyou for your review. I'll make sure to add proper comments in such cases.
